src/main/java/software/amazon/smithy/lsp/SmithyTextDocumentService.java

```diff
@@ -225,33 +230,112 @@ private File fileUri(TextDocumentItem tdi) {
     }
 
     /**
-     * @param path     Path of new model file.
-     * @param original Original model file to compare against when recompiling.
+     * @param result Either a fatal error message, or a list of diagnostics to
+     *               publish
      */
-    public void recompile(File path, Optional<File> original) {
-        Either<Exception, SmithyProject> loadedModel = this.project.recompile(path);
-        String changedFileUri = original.map(File::getAbsolutePath).orElse(path.getAbsolutePath());
-
+    public void report(Either<String, List<PublishDiagnosticsParams>> result) {
```

Comment:
Reporting and recompilation are now separate, to aid with testing and general declutter.